android-components/components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -535,37 +595,125 @@ data class ContextMenuCandidate(
                     additionalValidation(tab, hitResult)
             },
             action = { _, hitResult ->
-                clipPlaintText(
+                clipPlainText(
                     context,
                     hitResult.getLink(),
                     hitResult.src,
                     R.string.mozac_feature_contextmenu_snackbar_link_copied,
-                    snackBarParentView,
+                    snackbarParentView,
                     snackbarDelegate,
                 )
             },
         )
 
         @Suppress("LongParameterList")
-        private fun clipPlaintText(
+        private fun clipPlainText(
             context: Context,
             label: String,
             plainText: String,
             displayTextId: Int,
-            snackBarParentView: View,
+            snackbarParentView: View,
             snackbarDelegate: SnackbarDelegate = DefaultSnackbarDelegate(),
         ) {
-            val clipboardManager =
-                context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
             val clip = ClipData.newPlainText(label, plainText)
-            clipboardManager.setPrimaryClip(clip)
+            context.getClipboardManager().setPrimaryClip(clip)
 
             snackbarDelegate.show(
-                snackBarParentView = snackBarParentView,
+                snackbarParentView = snackbarParentView,
                 text = displayTextId,
                 duration = Snackbar.LENGTH_SHORT,
             )
         }
+
+        @OptIn(DelicateCoroutinesApi::class)
+        @Suppress("LongParameterList")
+        private fun clipImage(
+            context: Context,
+            dispatcher: CoroutineDispatcher,
+            imageSrc: String,
+            displayTextId: Int,
+            snackbarParentView: View,
+            snackbarDelegate: SnackbarDelegate = DefaultSnackbarDelegate(),
+            systemTimeProvider: () -> Long,
+            uriForFileProvider: (Context, File) -> Uri,
+        ) {
+            val imageUrl = try {
+                URL(imageSrc)
+            } catch (e: MalformedURLException) {
+                Log.log(Log.Priority.ERROR, ContextMenuCandidate::class.java.simpleName, e)
+                return
+            }
+
+            GlobalScope.launch(dispatcher) {
+                // Concatenate the current time (milliseconds) and the URL hash code (absolute value)
+                // to generate a file name.
+                val localTempFileName = context.getString(
+                    R.string.mozac_feature_contextmenu_copy_image_filename,
+                    systemTimeProvider.invoke().toString(),
+                    imageSrc.hashCode().absoluteValue.toString(),
+                )
+
+                val localTempFile = File(context.externalCacheDir, localTempFileName)
```

Comment:
I welcome any discussion around if this is the best place to store the 'temporary file'.